src/component.js

```diff
@@ -25,12 +25,19 @@ export function Component(props, context) {
  */
 Component.prototype.setState = function(update, callback) {
 	// only clone state when copying to nextState the first time.
-	let s =
-		(this._nextState !== this.state && this._nextState) ||
-		(this._nextState = assign({}, this.state));
+	let s;
+	if (this._nextState !== this.state) {
+		s = this._nextState;
+	}
+	else {
+		s = this._nextState = assign({}, this.state);
+	}
+
+	if (typeof update == 'function') {
+		update = update(s, this.props);
+	}
 
-	// if update() mutates state in-place, skip the copy:
-	if (typeof update!=='function' || (update = update(s, this.props))) {
+	if (update) {
```

Comment:
Original:
```js
("function" != typeof n || (n = n(u, this.props))) && s(u, n);
```

Updated:
```js
"function" == typeof n && (n = n(u, this.props));
n && s(u, n);
```

This change is probably a fluke as to how it saves us a byte. Perhaps the double use of `&&`? I think it be worth keeping cuz it expresses our intent using idiomatic JS and enables terser to minify this as it sees fit